pageserver/src/tenant_mgr.rs

```diff
@@ -350,6 +433,9 @@ pub fn detach_timeline(
                 .detach_timeline(timeline_id)
                 .context("Failed to detach inmem tenant timeline")?;
             tenant.local_timelines.remove(&timeline_id);
+            tenants_state::try_send_timeline_update(LocalTimelineUpdate::Detach(
```

Comment:
Feels like we'd better deal with both issues in https://github.com/neondatabase/neon/issues/1775 then?

Reply:
I'm ok with that, though the handling would be quite different. If I'm correct with my hypothesis for #1775 (I need to write a test to confirm that) it needs to take corresponding locks to ensure that compaction/gc are not running. And for the current problem here it requires some way to join the walreceiver which is a different story that this PR breaks.

Simpler option would be to pass a oneshot channel and let walrceiver send to it when it cleaned everything up. 

So up to you :) I'm ok with doing it in #1775 There shouldnt be any major problem from that because when data got deleted walreceiver will probably stop by itself because of the errors.